src/poetry/utils/authenticator.py

```diff
@@ -100,31 +212,57 @@ def request(self, method: str, url: str, **kwargs: Any) -> requests.Response:
                     raise e
             else:
                 if resp.status_code not in [502, 503, 504] or is_last_attempt:
-                    resp.raise_for_status()
+                    if resp.status_code is not None and raise_for_status:
+                        resp.raise_for_status()
                     return resp
 
             if not is_last_attempt:
                 attempt += 1
                 delay = 0.5 * attempt
-                self._log(f"Retrying HTTP request in {delay} seconds.", level="debug")
+                logger.debug(f"Retrying HTTP request in {delay} seconds.")
                 time.sleep(delay)
                 continue
 
         # this should never really be hit under any sane circumstance
         raise PoetryException("Failed HTTP {} request", method.upper())
 
-    def get_credentials_for_url(self, url: str) -> tuple[str | None, str | None]:
-        parsed_url = urllib.parse.urlsplit(url)
+    def get(self, url: str, **kwargs: Any) -> requests.Response:
+        return self.request("get", url, **kwargs)
 
-        netloc = parsed_url.netloc
+    def post(self, url: str, **kwargs: Any) -> requests.Response:
+        return self.request("post", url, **kwargs)
 
-        credentials: tuple[str | None, str | None] = self._credentials.get(
-            netloc, (None, None)
+    def _get_credentials_for_url(self, url: str) -> HTTPAuthCredential:
+        repository = self.get_repository_config_for_url(url)
+
+        credential = (
+            repository.get_http_credentials(password_manager=self._password_manager)
```

Comment:
How about adding `_get_credentiald_for_repository(self, repository)` which caches the result in `self._credentials[repository.url]`?
That should save keyring lookups when downloading multiple deps from the same repo.